tests/examples/source/devfiles/nodejs/devfile-deploy-ingress.yaml

```diff
@@ -0,0 +1,179 @@
+        isDefault: true
+        kind: debug
+      workingDir: /project
+    id: debug
+  - exec:
+      commandLine: npm test
+      component: runtime
+      group:
+        isDefault: true
+        kind: test
+      workingDir: /project
+    id: test
+# This is the main "composite" command that will run all below commands
+  - id: deploy
+    composite:
+      commands:
+        - build-image
+        - k8s-deployment
+        - k8s-service
+        - k8s-url
+      group:
+        isDefault: true
+        kind: deploy
+
+  # Below are the commands and their respective components that they are "linked" to deploy
+  - id: build-image
+    apply:
+      component: outerloop-build
+  - id: k8s-deployment
+    apply:
+      component: outerloop-deployment
+  - id: k8s-service
+    apply:
+      component: outerloop-service
+  - id: k8s-url
+    apply:
+      component: outerloop-url
+components:
+  - container:
+      endpoints:
+        - name: http-3000
+          targetPort: 3000
+      image: registry.access.redhat.com/ubi8/nodejs-14:latest
+      memoryLimit: 1024Mi
+      mountSources: true
+      sourceMapping: /project
+    name: runtime
+  # This will build the container image before deployment
+  - name: outerloop-build
+    image:
+      dockerfile:
+        buildContext: ${PROJECT_SOURCE}
+        rootRequired: false
+        uri: ./Dockerfile
+      imageName: "{{CONTAINER_IMAGE}}"
+
+  # This will create a Deployment in order to run your container image across
+  # the cluster.
+  - name: outerloop-deployment
+    kubernetes:
+      inlined: |
+        kind: Deployment
+        apiVersion: apps/v1
+        metadata:
+          name: {{RESOURCE_NAME}}
+        spec:
+          replicas: 1
+          selector:
+            matchLabels:
+              app: {{RESOURCE_NAME}}
+          template:
+            metadata:
+              labels:
+                app: {{RESOURCE_NAME}}
+            spec:
+              containers:
+                - name: {{RESOURCE_NAME}}
+                  image: {{CONTAINER_IMAGE}}
+                  ports:
+                    - name: http
+                      containerPort: {{CONTAINER_PORT}}
+                      protocol: TCP
+                  resources:
+                    limits:
+                      memory: "1024Mi"
+                      cpu: "500m"
+
+  # This will create a Service so your Deployment is accessible.
+  # Depending on your cluster, you may modify this code so it's a
+  # NodePort, ClusterIP or a LoadBalancer service.
+  - name: outerloop-service
+    kubernetes:
+      inlined: |
+        apiVersion: v1
+        kind: Service
+        metadata:
+          name: {{RESOURCE_NAME}}
+        spec:
+          ports:
+          - name: "{{CONTAINER_PORT}}"
+            port: {{CONTAINER_PORT}}
+            protocol: TCP
+            targetPort: {{CONTAINER_PORT}}
+          selector:
+            app: {{RESOURCE_NAME}}
+          type: ClusterIP
+  - name: outerloop-url
```

Comment:
Could be interesting to also add with an Ingress with no rules, but a [default backend](https://kubernetes.io/docs/concepts/services-networking/ingress/#default-backend), I think.

Reply:
In case a default backend is used, there will be no host, but if rules are given, we'll  use them.
No host, no rules: ingressName: */*
No host, rules: ingressName: */testpath